flaml/model.py

```diff
@@ -2287,36 +2288,47 @@ def fit(self, X_train, y_train, budget=None, **kwargs):
             monitor="val_loss", min_delta=1e-4, patience=10, verbose=False, mode="min"
         )
         lr_logger = LearningRateMonitor()  # log the learning rate
-        logger = TensorBoardLogger(
-            kwargs.get("log_dir", "lightning_logs")
-        )  # logging results to a tensorboard
-        default_trainer_kwargs = dict(
-            gpus=kwargs.get("gpu_per_trial", [0])
-            if torch.cuda.is_available()
-            else None,
-            max_epochs=max_epochs,
-            gradient_clip_val=gradient_clip_val,
-            callbacks=[lr_logger, early_stop_callback],
-            logger=logger,
-        )
-        trainer = pl.Trainer(
-            **default_trainer_kwargs,
-        )
-        tft = TemporalFusionTransformer.from_dataset(
-            training,
-            **params,
-            lstm_layers=2,  # 2 is mostly optimal according to documentation
-            output_size=7,  # 7 quantiles by default
-            loss=QuantileLoss(),
-            log_interval=10,  # uncomment for learning rate finder and otherwise, e.g. to 10 for logging every 10 batches
-            reduce_on_plateau_patience=4,
-        )
-        # fit network
-        trainer.fit(
-            tft,
-            train_dataloaders=train_dataloader,
-            val_dataloaders=val_dataloader,
-        )
+
+        def _fit(log):
+            default_trainer_kwargs = dict(
+                gpus=kwargs.get("gpu_per_trial", [0])
+                if torch.cuda.is_available()
+                else None,
+                max_epochs=max_epochs,
+                gradient_clip_val=gradient_clip_val,
+                callbacks=[lr_logger, early_stop_callback],
+                logger=log,
+            )
+            trainer = pl.Trainer(
+                **default_trainer_kwargs,
+            )
+            tft = TemporalFusionTransformer.from_dataset(
+                training,
+                **params,
+                lstm_layers=2,  # 2 is mostly optimal according to documentation
+                output_size=7,  # 7 quantiles by default
+                loss=QuantileLoss(),
+                log_interval=10,
+                # uncomment for learning rate finder and otherwise, e.g. to 10 for logging every 10 batches
+                reduce_on_plateau_patience=4,
+            )
+            # fit network
+            trainer.fit(
+                tft,
+                train_dataloaders=train_dataloader,
+                val_dataloaders=val_dataloader,
+            )
+            return trainer
+
+        try:
+            logger = TensorBoardLogger(
+                kwargs.get("log_dir", "lightning_logs")
+            )  # logging results to a tensorboard
+            trainer = _fit(log=logger)
+        except (ValueError, AttributeError):
+            # issue with pytorch forecasting model log_prediction() function
+            # pytorch-forecasting issue #1145
+            trainer = _fit(log=False)

```

Comment:
AttributeError was raised in this line instead of line 2327. I guess that means `log=False` doesn't work.
If you confirm that, and can't find a way to make it work, then this line should be removed. Instead, issue a warning that an error occurred in pytorch-forecasting. In `.predict()`, make a dumb prediction for such cases.

Reply:
issue was had to set `callbacks=False` as well